leaf/extensions/tags.py

```diff
@@ -0,0 +1,351 @@
+from bot import LeafBot
+from utils import Paginator
+from discord.ext import commands
+from discord import app_commands
+import discord
+import pytz
+
+from typing import Optional
+
+
+@app_commands.guild_only()
+class TagsCog(commands.GroupCog, name="Tags", group_name="tags"):
+    def __init__(self, bot: LeafBot) -> None:
+        self.bot = bot
+
+    @app_commands.describe(starting_page="The page to start on.")
+    @app_commands.command(name="list", description="Lists all the tags in the server.")
+    async def list_tags(
+        self,
+        interaction: discord.Interaction,
+        starting_page: Optional[int] = 1,
```

Comment:
This simply provides a second way to select the page alongside the button on the paginator. I doubt it will get used often but there is no easy way to skip forward by 5 or 10 pages currently, so maybe some improvements need to be made here.